### PR TITLE
Upgrade octicons

### PR DESCRIPTION
**Pull request by emplums**
🚨 Do not merge until installing canary doctocat version 🚨 

## Description
This PR upgrades `@primer/octicons-react` to version `10.0.0`. Since the new version of `octicons-react` removes the `Octicon` export, a couple of components needed to be refactored:

- CircleBadge
- CircleOcticon
- StyledOcticon
- TextInput

## Notes
Our original strategy was to use `@primer/styled-octicons` but after weighing the pros and cons a bit more, we've decided to stick with the original package + a `StyledOcticon` component. Please see the [closed issue for more details on this decision](https://github.com/primer/components/pull/774#issuecomment-652675898). I'm also going to hold off on marking it as a peer dependency for now.

Note that `CircleOcticon` looks janky since the size of the icon isn't proportional to the circle. This bug existed before these changes and needs to be addressed separately.

## 🚨 Breaking Changes 🚨 
- Though not directly related to changes in this repo, the new octicons version has significant API changes [see the release notes for the icons package for more details](https://github.com/primer/octicons/releases/tag/v10.0.0).

### CODE REVIEW

**Comment by emplums** [comment]
To do: double check that the `size` prop works properly

**Comment by emplums** [comment]
@BinaryMuse any idea why the `LabelGroup` test is failing? I didn't touch it in this PR and confirmed that the `sx` prop does work on `LabelGroup` 🤔 
image

I think `LabelGroup` [doesn't need this style wrapper anymore](https://github.com/primer/components/blob/master/src/LabelGroup.js#L18) and I thought maybe that had something to do with the `sx` prop not working, but refactoring didn't make the test pass either

File: src/StyledOcticon.js [hunk 1]
```diff
@@ -1,23 +1,34 @@
...
 StyledOcticon.defaultProps = {
-  theme
+  theme,
+  size: 16
 }
 
 StyledOcticon.propTypes = {
   ...COMMON.propTypes,
   ...sx.propTypes,
+  icon: PropTypes.element,
```
Comment (BinaryMuse) [inline src/StyledOcticon.js:27]: I think this should be `PropTypes.elementType`, and should be required.

```suggestion
  icon: PropTypes.elementType.isRequired,
```

File: index.d.ts [hunk 2]
```diff
@@ -133,8 +133,10 @@ declare module '@primer/components' {
   export const CircleOcticon: React.FunctionComponent<CircleOcticonProps>
 
   export interface StyledOcticonProps extends CommonProps {
-    size?: number
+    size?: number | 'small' | 'medium' | 'large'
     icon: React.ReactNode
```
Comment (BinaryMuse) [inline index.d.ts:137]: I think this also needs to be changed, but I'm not sure to what. Type `prop-types` DefinitelyTyped package defines a `ReactComponentLike` which [they use for `PropTypes.elementType`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/68523d90957c0568592f7bea00cde8a809063fbc/types/prop-types/index.d.ts#L68).

```typescript
export type ReactComponentLike =
    | string
    | ((props: any, context?: any) => any)
    | (new (props: any, context?: any) => any);
```

**Comment by BinaryMuse** [comment]
@emplums `CircleOcticon` also needs its `icon` prop updated

**Comment by emplums** [comment]
@BinaryMuse I made the changes you requested + tested that `ReactComponentLike` works well in a TS environment here: https://codesandbox.io/s/primer-components-ts-testing-47dbw?file=/src/index.tsx

**Pull request merged**